### PR TITLE
feat(item-properties): Add files count for folder in item properties

### PR DESCRIPTION
**Pull request by bkepka-box**
Changes:
- Add files count under size for folder in item properties
- Rewrite `src/features/item-details/__tests__/ItemProperties.test.js` from Enzyme to RTL

Screenshot 2025-09-23 at 13 30 02

## Summary by CodeRabbit

- **New Features**
  - Item details sidebar shows a localized files count next to size for folder items, with correct singular/plural and zero handling.
- **Tests**
  - Expanded unit and visual tests for item details (size/files-count, folder vs non-folder, singular/plural/zero, editable vs read-only fields); replaced snapshots with explicit DOM assertions; added a visual story for the details tab.
- **Documentation**
  - Added a localized message key for the files count label.

### CODE REVIEW

File: src/features/item-details/ItemProperties.js [hunk 1]
```diff
@@ -108,7 +111,12 @@ const ItemProperties = ({
             {size ? (
                 <>
                     <FormattedMessage tagName="dt" {...messages.size} />
-                    <dd>{size}</dd>
+                    <dd>
+                        <div>{size}</div>
+                        {filesCount !== null && type === ITEM_TYPE_FOLDER ? (
+                            <FormattedMessage values={{ filesCount }} {...messages.filesCountLabel} />
+                        ) : null}
+                    </dd>
```
[the remark this answers is not in the document]
Comment (bkepka-box) [reply src/features/item-details/ItemProperties.js:120]: I adjusted the code as you suggested

File: src/features/item-details/ItemProperties.js [hunk 2]
```diff
@@ -108,7 +111,12 @@ const ItemProperties = ({
             {size ? (
```
Comment (JChan106) [inline src/features/item-details/ItemProperties.js:112]: May be out of scope, but if size is 0, wouldn't this be hidden? Is that what we want?

File: src/features/item-details/ItemProperties.js [hunk 3]
```diff
@@ -108,7 +111,12 @@ const ItemProperties = ({
...
                 <>
                     <FormattedMessage tagName="dt" {...messages.size} />
-                    <dd>{size}</dd>
+                    <dd>
+                        <div>{size}</div>
```
Comment (JChan106) [inline src/features/item-details/ItemProperties.js]: could we use BP Text here?

File: src/features/item-details/__tests__/ItemProperties.test.js [hunk 4]
```diff
@@ -1,18 +1,54 @@
 import * as React from 'react';
+import { render, screen } from '../../../test-utils/testing-library';
+import { ITEM_TYPE_FOLDER, ITEM_TYPE_FILE } from '../../../common/constants';
 
 import ItemProperties from '../ItemProperties';
 
+jest.mock('../EditableDescription', () => {
```
Comment (JChan106) [inline src/features/item-details/__tests__/ItemProperties.test.js]: Cursor may be overaggressive in mocking, do all of these components need to be mocked for the tests to work properly? I would only mock if there are errors or the subcomponents are too complex

File: src/features/item-details/ItemProperties.js (hunk 2 above)
Comment (bkepka-box) [reply src/features/item-details/ItemProperties.js:112]: It won't be hidden because size is a string, for 0 it will be "0 B"

File: src/features/item-details/__tests__/ItemProperties.test.js (hunk 4 above)
Comment (bkepka-box) [reply src/features/item-details/__tests__/ItemProperties.test.js]: With the mocks it is easier to target specific components by `data-testid` to verify their appearance. As this is unit test I do not think it makes a big difference if we use real components. What do you think?

Comment (JChan106) [reply src/features/item-details/__tests__/ItemProperties.test.js]: We'e had some discussions around this in the team, I think if the actual components are easy to target, there's more benefit and confidence in our tests with keeping them rather than mocking. Not saying we shouldn't have mocks at all, but rather using mocks as a secondary option rather than default option. Just don't want to get to a point where everything in a component is mocked, and some edge case where the child component modifies the parent, and a regression slips through.

Comment (bkepka-box) [reply src/features/item-details/__tests__/ItemProperties.test.js]: Agree, I removed mocked components

**Review by loonskai — APPROVED** [review]

File: src/features/item-details/__tests__/ItemProperties.test.js [hunk 5]
```diff
@@ -1,18 +1,34 @@
...
-    test('should not render properties when not specified', () => {
-        const wrapper = getWrapper();
+    test('should render empty properties list and RetentionPolicy when no properties are specified', () => {
+        renderComponent({
+            retentionPolicyProps: {
+                retentionPolicyDescription: 'Retention policy description',
+            },
+        });
+
+        const propertiesList = document.querySelector('.item-properties');
```
Comment (greg-in-a-box) [inline src/features/item-details/__tests__/ItemProperties.test.js]: are we not able to use screen to target this element?

**Review by greg-in-a-box — CHANGES_REQUESTED** [review]
I think we need to add a VRT for the detail tab for the content-sidebar because there doesnt seem to be one that shows most of the fields and since this is changing one of the fields and we only have the unit tests, i highly suggest adding at least one story VRT.  you can copy one of them and navigate to the details tab

File: src/features/item-details/__tests__/ItemProperties.test.js (hunk 5 above)
Comment (bkepka-box) [reply src/features/item-details/__tests__/ItemProperties.test.js]: updated

File: src/features/item-details/ItemProperties.js [hunk 6]
```diff
@@ -31,14 +33,16 @@ const ItemProperties = ({
...
     trashedAt,
+    type,
     uploader,
     url,
 }) => {
     const descriptionId = uniqueid('description_');
 
     return (
-        <dl className="item-properties">
+        <dl className="item-properties" data-testid="item-properties">
```
Comment (greg-in-a-box) [inline src/features/item-details/ItemProperties.js:45]: do we still need this? should be able to target the dl https://stackoverflow.com/a/76826101

File: src/elements/content-sidebar/__tests__/__snapshots__/SidebarFileProperties.test.js.snap [hunk 7]
```diff
@@ -118,6 +118,7 @@ exports[`elements/content-sidebar/SidebarFileProperties render() should render r
       >
         <dl
```
Comment (greg-in-a-box) [inline src/elements/content-sidebar/__tests__/__snapshots__/SidebarFileProperties.test.js.snap:119]: we should be able to target this with getByRole 
https://stackoverflow.com/a/76826101

**Review by greg-in-a-box — CHANGES_REQUESTED** [review]

File: src/features/item-details/ItemProperties.js (hunk 6 above)
Comment (bkepka-box) [reply src/features/item-details/ItemProperties.js:45]: Based on the link you provided `dl` does not have implicit ARIA role which we could target. I think `data-testid` is necessary there if we want to keep the check if `item-properties` are rendered.

File: src/elements/content-sidebar/__tests__/__snapshots__/SidebarFileProperties.test.js.snap (hunk 7 above)
Comment (bkepka-box) [reply src/elements/content-sidebar/__tests__/__snapshots__/SidebarFileProperties.test.js.snap:119]: same as above

File: src/features/item-details/ItemProperties.js (hunk 6 above)
Comment (greg-in-a-box) [reply src/features/item-details/ItemProperties.js:45]: whoops misread dt for dl

File: src/elements/content-sidebar/stories/tests/ContentSidebar-visual.stories.tsx [hunk 8]
```diff
@@ -36,3 +36,10 @@ export const ContentSidebar: StoryObj<typeof ContentSidebarComponent> = {
         },
     },
 };
+
+export const ContentSidebarDetailsTab: StoryObj<typeof BoxAISidebar> = {
```
Comment (greg-in-a-box) [inline src/elements/content-sidebar/stories/tests/ContentSidebar-visual.stories.tsx]: `BoxAISidebar` doesnt exist, where are we importing it and why arent we using `ContentSidebarComponent` instead?

Comment (bkepka-box) [reply src/elements/content-sidebar/stories/tests/ContentSidebar-visual.stories.tsx]: Yeah sorry, accidentally left that. I've fixed it now

**Review by loonskai — APPROVED** [review]

**Review by greg-in-a-box — APPROVED** [review]
